lib/mozilla-ast.js

```diff
@@ -1158,11 +1163,18 @@
 
     var FROM_MOZ_STACK = null;
 
-    function from_moz(node) {
-        FROM_MOZ_STACK.push(node);
-        var ret = node != null ? MOZ_TO_ME[node.type](node) : null;
+    function from_moz(moz) {
+        FROM_MOZ_STACK.push(moz);
+        var node = null;
+        if (moz) {
+            if (!HOP(MOZ_TO_ME, moz.type)) {
+                var s = my_start_token(moz);
+                js_error("Unsupported type: " + moz.type, s.file, s.line, s.col, s.pos);
```

Comment:
I tried that too - the `s.file, s.line, s.col, s.pos` part is useless for mozilla nodes. Only `moz.start` and `moz.end` positions are available.  The other use of `js_error` in this source file may also be wrong.

Reply:
Yeah, probably should just stick to `throw new Error()` like most other places in this file...